### DISCUSSION TITLE
Set `cb_eval` after context creation

### DISCUSSION THREAD

**Discussion by vladfaust** [ggml-org/llama.cpp#8284] [Ideas]
I'd like to be able to change `cb_eval` and `cb_eval_user_data` prior to calling `llama_decode`, per-call, to track decoding progress with different callbacks. Currently it's a part of `llama_context.cparams`. Exposing the whole `.cparams` field doesn't seem to be viable, but changing these two is required in my use-case. Any workarounds?

**Comment by ggerganov** [comment]
You can have a wrapper callback that calls the callback that you want
